pyblish_nuke/lib.py

```diff
@@ -260,31 +260,28 @@ def dock(window):
     # There is a bug where existing docks are kept in-memory when closed via UI
     if self._dock:
         print "Deleting existing dock..."
-        self._dock.layout().addWidget(window)
-
         parent = self._dock
         dialog = None
         stacked_widget = None
         main_windows = []
 
         # Getting dock parents
         while parent:
-            if parent.__class__.__name__ == "QDialog":
+            if isinstance(parent, QtWidgets.QDialog):
                 dialog = parent
-            if parent.__class__.__name__ == "QStackedWidget":
+            if isinstance(parent, QtWidgets.QStackedWidget):
                 stacked_widget = parent
-            if parent.__class__.__name__ == "QMainWindow":
+            if isinstance(parent, QtWidgets.QMainWindow):
                 main_windows.append(parent)
             parent = parent.parent()
 
         dialog.deleteLater()
 
-        # If there are more than one main window, its a floating window
         if len(main_windows) > 1:
-            # If the stacked widget only contains 1 widget, its empty
-            # and we can close the empty floating window.
-            # This is natural Nuke UI behaviour.
+            # Then it's a floating window
             if stacked_widget.count() == 1:
+                # Then it's empty and we can close it,
```

Comment:
I hope this read better to not just me, but to you as well? Otherwise there's little point in changing it I think.